tests/functional/event_handler/test_api_gateway.py

```diff
@@ -1197,6 +1197,26 @@ def base():
     assert result["multiValueHeaders"]["Content-Type"] == [content_types.APPLICATION_JSON]
 
 
+def test_api_gateway_app_with_strip_prefix_and_route_prefix():
+    # GIVEN a strip_prefix matches the request path
+    # and a router instance
+    app = ApiGatewayResolver(strip_prefixes=["/foo"])
+    router = Router()
+    event = {"httpMethod": "GET", "path": "/foo/bar/users", "resource": "/users"}
+
+    @router.get("/users")
+    def base():
+        return {}
+
+    # THEN we can register the router with a prefix
+    app.include_router(router, prefix="/bar")
+
+    result = app(event, {})
+
+    # THEN process event correctly
+    assert result["statusCode"] == 200
```

Comment:
```suggestion
    # GIVEN all routes are stripped from its version e.g., /v1

    app = ApiGatewayResolver(strip_prefixes=["/v1"])
    router = Router()

    event = {"httpMethod": "GET", "path": "/v1/users/leandro", "resource": "/users"}

    @router.get("<user_id>")
    def base(user_id: str):
        return {"user": user_id}

    # WHEN a router is included prefixing all routes with "/users"
    app.include_router(router, prefix="/users")

    result = app(event, {})

    # THEN route correctly to the registered route after stripping each prefix (global + router)
    assert result["statusCode"] == 200
```

Two changes to make it easier to document its behaviour we're testing

1. Paths to be more realistic using API versioning `/v1` and service specific routes that are often split `/users`
2. Adjust GIVEN WHEN THEN to reflect this accordingly

Reply:
Yeah! Thanks for the feedback :heart: